pkgs/applications/networking/browsers/mullvad-browser/default.nix

```diff
@@ -88,10 +89,20 @@ let
       ++ lib.optionals pulseaudioSupport [ libpulseaudio ]
       ++ lib.optionals libvaSupport [ libva ]
       ++ lib.optionals mediaSupport [ ffmpeg ]
-  );
+  ) else "";
 
   version = "13.0.9";
 
+  # MacOS sources
+  darwinSource = "https://cdn.mullvad.net/browser/${version}/mullvad-browser-macos-${version}.dmg";
```

Comment:
What about the other mirrors that were used for the linux package?
I think they should also all distribute the macOS package.

Perhaps we should streamline the source urls this into a function for all platforms.

Reply:
Does this solution seem sensible? https://github.com/NixOS/nixpkgs/pull/283874/commits/fa7ff3c54ae65c205b4e03ee43fe0890640f2005#diff-1bd78721e31ae529d71193e63f92f114354272ba99d60406026dbfb41799b27aR97-R109

Not sure how this will impact/cooperate with the tor browser update.nix

Reply:
I'm not entirely sure about the use of `stdenv` here as that would probably mean `sources` can't be evaluated on any platform anymore. Just to go sure, instead of relying on `stdenv` perhaps just turn it into a function:

```nix
mkUrls = platform: extension: [
  "https://cdn.mullvad.net/browser/${version}/mullvad-browser-${platform}-${version}.${extension}"
  "https://github.com/mullvad/mullvad-browser/releases/download/${version}/mullvad-browser-${platform}-${version}.${extension}"
  "https://archive.torproject.org/tor-package-archive/mullvadbrowser/${version}/mullvad-browser-${platform}-${version}.${extension}"
  "https://dist.torproject.org/mullvadbrowser/${version}/mullvad-browser-${platform}-${version}.${extension}"
  "https://tor.eff.org/dist/mullvadbrowser/${version}/mullvad-browser-${platform}-${version}.${extension}"
  "https://tor.calyxinstitute.org/dist/mullvadbrowser/${version}/mullvad-browser-${platform}-${version}.${extension}"
];
```

Reply:
Thank you for the example! Will implement tomorrow 👍

Reply:
Does this work?
https://github.com/NixOS/nixpkgs/blob/46c2609eba835c898d23e3f06097e4b1cdf7c941/pkgs/applications/networking/browsers/mullvad-browser/default.nix#L97-L119


```
  mkUrls = platform: extension: [
    "https://cdn.mullvad.net/browser/${version}/mullvad-browser-${platform}-${version}.${extension}"
    "https://github.com/mullvad/mullvad-browser/releases/download/${version}/mullvad-browser-${platform}-${version}.${extension}"
    "https://archive.torproject.org/tor-package-archive/mullvadbrowser/${version}/mullvad-browser-${platform}-${version}.${extension}"
    "https://dist.torproject.org/mullvadbrowser/${version}/mullvad-browser-${platform}-${version}.${extension}"
    "https://tor.eff.org/dist/mullvadbrowser/${version}/mullvad-browser-${platform}-${version}.${extension}"
    "https://tor.calyxinstitute.org/dist/mullvadbrowser/${version}/mullvad-browser-${platform}-${version}.${extension}"
  ];

  sources = {
    x86_64-linux = fetchurl {
      urls = mkUrls "linux-x86_64" "tar.xz";
      hash = "sha256-TAtBlSkfpqsROq3bV9kwDYIJQAXSVkwxQwj3wIYEI7k=";
    };
    aarch64-darwin = fetchurl {
      urls = mkUrls "macos" "dmg";
      hash = "sha256-1rXzXLbmBd567MCUNGL1Vw0xfpk4WdoAG6weJgz9+Pg=";
    };
    x86_64-darwin = fetchurl {
      urls = mkUrls "macos" "dmg";
      hash = "sha256-1rXzXLbmBd567MCUNGL1Vw0xfpk4WdoAG6weJgz9+Pg=";
    };
  };
```

Reply:
That's exactly what I imagined.